contributor-docs/adrs/adr-021-automated-prop-docs.md

```diff
@@ -0,0 +1,277 @@
+
+### Comparing third-party tools
+
+I explored the following tools to help us extract component API data:
+
+- [react-docgen-typescript](https://github.com/styleguidist/react-docgen-typescript)
+- [TypeDoc](https://typedoc.org/)
+- [react-docgen](https://react-docgen.dev/)
+- [api-extractor](https://api-extractor.com/)
+
+Alternatively, we could go lower-level and build our own tool from scratch. I have not yet attempted this. See the following examples:
+
+- [Adobe Spectrum DocsTransformer](https://github.com/adobe/react-spectrum/blob/main/packages/dev/parcel-transformer-docs/DocsTransformer.js)
+- [MUI ComponentApiBuilder](https://github.com/mui/material-ui/blob/next/packages/api-docs-builder/ApiBuilders/ComponentApiBuilder.ts)
+
+I found [react-docgen-typescript](https://github.com/styleguidist/react-docgen-typescript) and [TypeDoc](https://typedoc.org/) to be the most viable tools to consider.
+
+Currently, I'd prefer to move forward with [react-docgen-typescript](https://github.com/styleguidist/react-docgen-typescript) because it has given me the best results so far, and it seems possible to customize and improve to better suit our needs.
+
+#### [react-docgen-typescript](https://github.com/styleguidist/react-docgen-typescript)
+
+##### react-docgen-typescript pros
+
+- Made specifically for extracting type data from React components
+- I'm able to get all components and prop data parsed into a `components.json` file (example of existing JSON file: [components.json](https://unpkg.com/browse/@primer/react@36.27.0/generated/components.json))
+- The `shouldIncludeExpression` option opens us up to further customization because it has the option to return expression data with output
+- Is the same tool Storybook uses to show prop types, so Storybook will be better aligned with component docs
+
+##### react-docgen-typescript cons
+
+- Requires some minor changes to source code to be able to parse all components and their props. The biggest change is that a component function needs to be exported in order to be picked up
+- Releases and contributions have slowed down in the repo
+- Having trouble handling a file with multiple components and multiple `.displayName` assignments
+- Props on components cast using `as PolymorphicForwardRefComponent<El, Props>` are not picked up
+- I can't just point to `src/index.ts`, `src/drafts/index.ts`, and `src/deprecated/index.ts` to only get exported components and hooks. Doing this only gets some components parsed. Instead, I have to parse all `.tsx` files (excluding `*.story.tsx` and `*.test.tsx`)
+- Cannot handle subcomponents like ActionList that are exported like this: `export const ActionList = Object.assign(List, Item: ActionListItem, etc...)`
+
+##### Example component
+
+`primer-react-docgen` and other tools I've tested only document components and props that are exported.
```

Comment:
Would this be `react-docgen` or `react-typescript-docgen`?

```suggestion
`react-typescript-docgen` and other tools I've tested only document components and props that are exported.
```

Reply:
`react-typescript-docgen`. `react-docgen` didn't do a very good job...